### PR TITLE
GameBoard CivilisationCards

### LINKED ISSUE #3: AllPlayresTakeAReward class
Unit tests: Yes. Focus on tryToMakeAutomaticAction method.

### LINKED ISSUE #15: GetCard class
Represents corresponding immediate civilisation card effect. Uses CurrentThrow to perform the effect

Unit tests: Yes.

### LINKED ISSUE #16: GetSomethingChoice class
Represents corresponding immediate civilisation card effect. Uses CurrentThrow to perform the effect

Unit tests: Yes.

### LINKED ISSUE #17: GetSomethingFixed class
Represents corresponding immediate civilisation card effect. Uses CurrentThrow to perform the effect

Unit tests: Yes.

### LINKED ISSUE #18: GetSomethingThrow class
Represents corresponding immediate civilisation card effect. Uses CurrentThrow to perform the effect

Unit tests: Yes.

### LINKED ISSUE #20: RewardMenu class
Represents the choices available in All Players Take A Reward imediate civilisation card effect.

Unit tests: Yes.

### LINKED ISSUE #23: Throw class
Encapsulates the random effect.

Unit tests: No. Class needs to be mocked in integration tests

### PR DESCRIPTION
**Pull request by johankogoodvibes**
resolves #23 
resolves #16 
resolves #17 
resolves #18 
resolves #20 
resolves #15 
resolves #3 
resolves #15

### CODE REVIEW

File: src/main/java/sk/uniba/fmph/dcs/game_board/RewardMenu.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+     */
+    @Override
+    public HasAction tryMakeAction(final PlayerOrder player) {
+        if (menu.isEmpty()) {
+            return HasAction.NO_ACTION_POSSIBLE;
+        }
+        if (!playersLeft.contains(player)) {
+            return HasAction.NO_ACTION_POSSIBLE;
+        }
+        return HasAction.WAITING_FOR_PLAYER_ACTION;
```
Comment (Stanko2) [inline src/main/java/sk/uniba/fmph/dcs/game_board/RewardMenu.java:103]: If `player` is the last player left perform the action automatically

File: src/main/java/sk/uniba/fmph/dcs/game_board/GetSomethingFixed.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+    /**
+     * take one resource and give to player.
+     *
+     * @param player player to give choice to
+     * @param choice resource chosen
+     * @return false if resource is not available to choose
+     */
+    @Override
+    public boolean performEffect(final Player player, final Effect choice) {
```
Comment (Stanko2) [inline src/main/java/sk/uniba/fmph/dcs/game_board/GetSomethingFixed.java:26]: Ignore choice and put everything to player

**Review by Stanko2 — CHANGES_REQUESTED** [review]

**Review by Stanko2 — APPROVED** [review]

**Comment by Duchyna1** [comment]
@johankogoodvibes @Stanko2 môžem mergnut? Či ešte niečo?

**Comment by johankogoodvibes** [comment]
Asi hej

**Pull request merged**